### PR TITLE
[codex] Sanitize empty optional MCP arguments

### PR DESCRIPTION
**Pull request by Tobi1chi**
## Summary
- sanitize MCP tool arguments before dispatch so empty optional payload values are omitted
- preserve the reconnect-and-retry path while sending cleaned arguments to the MCP session
- add debug logging so sanitized payload changes are visible when tracing tool calls

## Why
Some MCP servers reject requests when optional arguments are present but empty. This change strips empty placeholder values before the tool call so those optional fields are omitted instead of being sent as invalid payload.

## Validation
- `./.venv-precommit/bin/pre-commit run --from-ref upstream/master --to-ref HEAD`

## Summary by Sourcery

Sanitize MCP tool call arguments to strip empty optional values before dispatch and log any changes while preserving the reconnect-and-retry behavior.

Bug Fixes:
- Avoid MCP tool call failures by removing empty optional arguments and nested empty collections from the payload before sending requests.

Enhancements:
- Add debug logging to surface how MCP tool arguments are transformed during sanitization.

Tests:
- Add unit tests for MCP argument sanitization and for falling back to empty top-level arguments in MCP tool calls.

### CODE REVIEW

File: astrbot/core/agent/mcp_client.py [hunk 1]
```diff
@@ -125,6 +125,38 @@ async def _quick_test_mcp_connection(config: dict) -> tuple[bool, str]:
...
+        return cleaned_items if cleaned_items else _EMPTY_MCP_ARGUMENT
+
+    if isinstance(value, dict):
+        cleaned_dict = {}
+        for key, item in value.items():
+            cleaned_item = _sanitize_mcp_arguments(item)
+            if cleaned_item is _EMPTY_MCP_ARGUMENT:
+                continue
+            cleaned_dict[key] = cleaned_item
+        return cleaned_dict
```
[the remark this answers is not in the document]
Comment (Tobi1chi) [reply astrbot/core/agent/mcp_client.py]: Updated in `4200317e`.

I kept the top-level fallback to `{}` because MCP tool calls still need an object payload when every argument is omitted, but nested empty collections are now handled consistently: empty dicts and empty lists both collapse to the sentinel and get removed from the parent payload.

I also added unit coverage for both cases:
- nested empty collections are removed from the payload
- a fully-sanitized top-level argument object still falls back to `{}` before dispatch

Validation:
- `uv run pytest tests/unit/test_mcp_client.py`
- `./.venv-precommit/bin/pre-commit run --files astrbot/core/agent/mcp_client.py tests/unit/test_mcp_client.py`

**Comment by Tobi1chi** [comment]
原issue: https://github.com/AstrBotDevs/AstrBot/issues/7116

**Comment by Tobi1chi** [comment]
@codex review this PR

File: astrbot/core/agent/mcp_client.py [hunk 2]
```diff
@@ -125,6 +125,38 @@ async def _quick_test_mcp_connection(config: dict) -> tuple[bool, str]:
...
+                continue
+            cleaned_items.append(cleaned_item)
+        return cleaned_items if cleaned_items else _EMPTY_MCP_ARGUMENT
+
+    if isinstance(value, dict):
+        cleaned_dict = {}
+        for key, item in value.items():
+            cleaned_item = _sanitize_mcp_arguments(item)
+            if cleaned_item is _EMPTY_MCP_ARGUMENT:
+                continue
```
[the remark this answers is not in the document]
Comment (Tobi1chi) [reply astrbot/core/agent/mcp_client.py:174]: @codex Solved this problem, please review it again

[the remark this answers is not in the document]
Comment (Tobi1chi) [reply astrbot/core/agent/mcp_client.py]: @codex review the newest commit
